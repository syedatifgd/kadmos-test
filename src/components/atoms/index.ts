export { default as Page } from "./Page";
export type { PageProps } from "./Page";
export { default as UserCard } from "./UserCard";
export { default as LoadingView } from "./LoadingView";
