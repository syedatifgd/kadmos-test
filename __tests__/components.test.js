import React from "react";
import renderer from "react-test-renderer";
import { Page, UserCard, LoadingView } from "../src/components/atoms";

describe("Components render correctly", () => {
  it("renders loading view component correctly", () => {
    const tree = renderer.create(<LoadingView/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders page component correctly", () => {
    const tree = renderer.create(<Page/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders usercard component correctly", () => {
    const tree = renderer.create(<UserCard/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
