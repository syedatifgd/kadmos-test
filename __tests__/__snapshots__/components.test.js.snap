// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components render correctly has 1 child 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "height": 80,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    size="small"
  />
</View>
`;

exports[`Components render correctly renders loading view component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "height": 80,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    size="small"
  />
</View>
`;

exports[`Components render correctly renders page component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`Components render correctly renders usercard component correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5A5A5A",
      "borderRadius": 10,
      "height": 120,
      "marginHorizontal": 10,
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "borderRadius": 10,
        "height": "100%",
        "overflow": "hidden",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 25,
            "fontWeight": "bold",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginLeft": 5,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "bottom": 5,
          "color": "white",
          "fontSize": 15,
          "fontWeight": "bold",
          "left": 10,
          "position": "absolute",
        }
      }
    />
  </View>
</View>
`;
